File: src/main/java/catwalks/gui/CatwalksModConfGuiFactory.java
package catwalks.gui;

import java.util.Set;

import net.minecraftforge.fml.client.IModGuiFactory;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiScreen;

public class CatwalksModConfGuiFactory implements IModGuiFactory
{
    @Override
    public void initialize(Minecraft minecraftInstance)
    {
    }

    @Override
    public Class<? extends GuiScreen> mainConfigGuiClass()
    {
        return CatwalksModConfGui.class;
    }

    @Override
    public Set<RuntimeOptionCategoryElement> runtimeGuiCategories()
    {
        return null;
    }

    @Override
    public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement element)
    {
        return null;
    }
}

File: src/main/java/catwalks/gui/CatwalksModConfGuiFactory.kt
package catwalks.gui

import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.client.IModGuiFactory

class CatwalksModConfGuiFactory : IModGuiFactory {
    override fun initialize(minecraftInstance: Minecraft) {
    }

    override fun mainConfigGuiClass(): Class<out GuiScreen> {
        return CatwalksModConfGui::class.java
    }

    override fun runtimeGuiCategories(): Set<IModGuiFactory.RuntimeOptionCategoryElement>? {
        return null
    }

    override fun getHandlerFor(element: IModGuiFactory.RuntimeOptionCategoryElement): IModGuiFactory.RuntimeOptionGuiHandler? {
        return null
    }
}